docker-compose.traefik.yml

```diff
@@ -0,0 +1,46 @@
+# For details and other explanations about this file refer to:
+# https://github.com/vrtmrz/obsidian-livesync/blob/main/docs/setup_own_server.md#traefik
+
+version: "2.1"
+services:
+  couchdb:
+    image: couchdb:latest
+    container_name: obsidian-livesync
+    user: 1000:1000
+    environment:
+      - COUCHDB_USER=username
+      - COUCHDB_PASSWORD=password
+    volumes:
+      - ./data:/opt/couchdb/data
+      - ./local.ini:/opt/couchdb/etc/local.ini
+    # Ports not needed when already passed to Traefik
+    #ports:
+    #  - 5984:5984
+    restart: unless-stopped
+    networks:
+      - proxy
+    labels:
+      - "traefik.enable=true"
+      # The Traefik Network
+      - "traefik.docker.network=proxy"
+      # Don't forget to replace 'obsidian-livesync.example.org' with your own domain
+      - "traefik.http.routers.obsidian-livesync.rule=Host(`obsidian-livesync.example.org`)"
+      # The 'websecure' entryPoint is basically your HTTPS entrypoint. Check the next code snippet if you are encountering problems only; you probably have a working traefik configuration if this is not your first container you are reverse proxying.
+      - "traefik.http.routers.obsidian-livesync.entrypoints=websecure"
+      - "traefik.http.routers.obsidian-livesync.service=obsidian-livesync"
+      - "traefik.http.services.obsidian-livesync.loadbalancer.server.port=5984"
+      - "traefik.http.routers.obsidian-livesync.tls=true"
+      # Replace the string 'letsencrypt' with your own certificate resolver
+      - "traefik.http.routers.obsidian-livesync.tls.certresolver=letsencrypt"
+      - "traefik.http.routers.obsidian-livesync.middlewares=obsidiancors"
+      # The part needed for CORS to work on Traefik 2.x starts here
+      - "traefik.http.middlewares.obsidiancors.headers.accesscontrolallowmethods=GET,PUT,POST,HEAD,DELETE"
```

Comment:
Does the `headers` middleware actually need? CORS is expected to be handled by CouchDB.
Could you please check the behaviour without this middleware?

Reply:
To my knowledge and experience it is **definitely** required, a default Traefik config won't work without these headers allowed [explicitly](https://doc.traefik.io/traefik/middlewares/http/headers/#cors-headers). I tested it with my production and local test server. Someone can also define these 'middlewares' inside other files too and [reference it](https://doc.traefik.io/traefik/providers/overview/#provider-namespace) inside Traefik's docker-compose.yml file, but for the sake of simplicity having all the Traefik labels in the docker-compose.yml file is better for getting the job done faster; generally working with one file in this case.

It may look weird at first that we have to define middlewares for allowing CORS and AccessControlAllowCredentials headers to be passed, but this is the Traefik way and it works reliably.

Reply:
Thank you for the detail! I have understood the behaviour of traefik. And I would love to apologise for my misunderstanding and withdraw this point.

I have not understood traefik on Docker well. This will enrich our knowledge!
Everything has got clear. Thank you very much!